test/decl/func/functions.swift

```diff
@@ -179,3 +179,9 @@ func parentheticalInout2(_ fn: (((inout Int)), Int) -> ()) {
   var value = 0
   fn(&value, 0)
 }
+
+// Function declaration with uninhabited parameter type
+func foo(baz: Never) -> Int { // expected-warning{{will never be executed}}
+  print("I can't be called!")
+  return 0
+}
```

Comment:
Does that mean that `func foo(baz: Never) -> Int {}` works now? If so, could you add a test for it? 😄

Reply:
It would already contain an `unreachable`. Here's the raw SIL for the above code:

```
// foo(baz:)
sil hidden @test1.foo(baz: Swift.Never) -> Swift.Int : $@convention(thin) (Never) -> Int {
// %0                                             // user: %1
bb0(%0 : $Never):
  debug_value %0 : $Never, let, name "baz", argno 1 // id: %1
  unreachable                                     // id: %2
} // end sil function 'test1.foo(baz: Swift.Never) -> Swift.Int'
```

Reply:
But it would complain when you invoke the frontend with `emit-ir` about a missing return statement. Let me give it a shot with my change so see if it still complains!

Reply:
Just tested and it works! Invoking the front-end with `-emit-ir` no longer complains about a missing return. I have updated the tests as well 👍